lib/storage-handler.js

```diff
@@ -253,6 +253,7 @@ exports.download = function(provider, req, res, container, file, cb) {
     reader.pipe(res);
 
     reader.on('error', function onReaderError(err) {
+      res.set('content-type', 'application/json');
```

Comment:
hard coding the `content-type` might be a bad idea because it might mean if your error handler returns a `text/html 404`, or anything else, the `download()'s onError` should not really make the assumption that the error must be in a form of `application/json`, and maybe we should perform content-negotiation with the `request's accept` to see what we set it as

where in our case `processError()` is the function that performs operation on the errorObject, so it might make sense to have such logic at the same place 

my suggestion was to modify the signature of processError, but im not sure if that make sense either